modules/network/src/main/java/org/apache/ignite/network/MessageMapper.java

```diff
@@ -0,0 +1,44 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ignite.network;
+
+import java.io.ObjectInputStream;
+import java.io.ObjectOutputStream;
+
+/**
+ * MessageMapper provides functionality for reading and writing {@link NetworkMessage} objects.
+ * @param <M>
+ */
+public interface MessageMapper<M extends NetworkMessage> {
+    /**
+     * Read
+     * @param objectInputStream
+     * @return
+     * @throws MessageMappingException
+     */
+    M readMessage(ObjectInputStream objectInputStream) throws MessageMappingException;
```

Comment:
Usage of ObjectInputStream is dictated by single implementation provided by ScaleCube. This is wrong, we should try to define an abstract API and hide any details in implementation.

Reply:
I agree, org.apache.ignite.plugin.extensions.communication.Message#writeTo and org.apache.ignite.plugin.extensions.communication.Message#readFrom signatures should be kept in mind.